File: 350. Intersection of Two Arrays II/solution.js
/**
 * @param {number[]} nums1
 * @param {number[]} nums2
 * @return {number[]}
 */
var intersect = function(nums1, nums2) {
  return nums1.filter(n => {
    const idx = nums2.indexOf(n);
    if (idx !== -1) {
      nums2.splice(idx, 1);
    }
    return idx !== -1;
  });
};

File: 350. Intersection of Two Arrays II/solution.py
class Solution:
    def intersect(self, nums1: List[int], nums2: List[int]) -> List[int]:
        dic1 = {}
        for i in nums1:
            dic1[i] = dic1.get(i, 0)+1
        out = []
        for i in nums2:
            if i in dic1 and dic1[i] > 0:
                out.append(i)
                dic1[i] -= 1
        return out
